src/software/ai/hl/stp/tactic/goalie/goalie_fsm.h

```diff
@@ -168,37 +197,49 @@ struct GoalieFSM
         DEFINE_SML_STATE(PivotKickFSM)
         DEFINE_SML_STATE(PositionToBlock)
         DEFINE_SML_STATE(MoveToGoalLine)
+        DEFINE_SML_STATE(DribbleFSM)
 
         DEFINE_SML_EVENT(Update)
 
-        DEFINE_SML_GUARD(ballInDefenseArea)
+        DEFINE_SML_GUARD(ballInInflatedDefenseArea)
         DEFINE_SML_GUARD(panicDone)
+        DEFINE_SML_GUARD(shouldEvacuateCrease)
         DEFINE_SML_GUARD(shouldPivotChip)
         DEFINE_SML_GUARD(shouldPanic)
         DEFINE_SML_GUARD(shouldMoveToGoalLine)
+        DEFINE_SML_GUARD(retrieveDone)
 
         DEFINE_SML_ACTION(panic)
         DEFINE_SML_ACTION(positionToBlock)
         DEFINE_SML_ACTION(moveToGoalLine)
         DEFINE_SML_SUB_FSM_UPDATE_ACTION(updatePivotKick, PivotKickFSM)
+        DEFINE_SML_SUB_FSM_UPDATE_ACTION(retrieveFromDeadZone, DribbleFSM)
 
         return make_transition_table(
             // src_state + event [guard] / action = dest_state
             *PositionToBlock_S + Update_E[shouldMoveToGoalLine_G] / moveToGoalLine_A =
                 MoveToGoalLine_S,
-            PositionToBlock_S + Update_E[shouldPanic_G] / panic_A = Panic_S,
+            PositionToBlock_S +
+                Update_E[shouldEvacuateCrease_G] / retrieveFromDeadZone_A = DribbleFSM_S,
+            PositionToBlock_S + Update_E[shouldPanic_G] / panic_A         = Panic_S,
             PositionToBlock_S + Update_E[shouldPivotChip_G] / updatePivotKick_A =
                 PivotKickFSM_S,
             PositionToBlock_S + Update_E / positionToBlock_A,
+            DribbleFSM_S + Update_E[retrieveDone_G] / updatePivotKick_A = PivotKickFSM_S,
```

Comment:
maybe we should have a transition from `DribbleFSM` to Panic_S in case an enemy comes nearby unless you have a good reason

Reply:
bump @germchow

Reply:
pending other comment below (accounting for enemy velocity)